### ISSUE TITLE
Nested triple backquotes cause spurious errors

### ISSUE THREAD

**Issue by anntzer** [PyCQA/pydocstyle#542]
Consider the following example:
```python
"""
A test module.

This is a module that implements an abstract base class Base, whose subclasses
must implement a ``frobnicate`` method, as follows::

    class Concrete(Base):
        def frobnicate(self, thing):
            \"""Frobnicate the thing.\"""

We want the code example to use triple quotes so that things appear correctly
in the sphinx-rendered docs.
"""
```
Running pydocstyle 6.1.1 on this module yields
```
/tmp/test.py:1 at module level:
        D300: Use """triple double quotes""" (found """-quotes)
/tmp/test.py:1 at module level:
        D301: Use r""" if any backslashes in a docstring
```
both of which seem wrong: I guess the first warning appears because pydocstyle gets confused by the inner triple quotes(?), and the second one is wrong as we actually want literal quotes to appear in the docstring.
